### PR TITLE
fixes logout error

### LINKED ISSUE #58: Weird error pops up on logout
It still logs you out, but...

screen shot 2017-08-31 at 11 33 26 am

### PR DESCRIPTION
**Pull request by Chauce**
Closes #58 

Just wrapped the whole thin in an `if` statement. Seems to have done the trick.

### CODE REVIEW

File: app/assets/javascripts/datatables.js [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-      { "bSortable": false },
-      null,
-      { "bSortable": false },
-      { "bSortable": false },
-      { "bSortable": false },
-    ],
-    "aLengthMenu": [[25, 50, 75, 100, -1], [25, 50, 75, 100, "All"]],
-    "iDisplayLength": 100
-  });
+  if ( !$.fn.dataTable.isDataTable( '#passengers' ) ) {
```
Comment (dfaulken) [inline app/assets/javascripts/datatables.js]: I know the place you copied from had space inside parentheses, but you don't need them (either set).

**Comment by dfaulken** [comment]
To clarify my comment, I suggest something like:

```js
$(document).on("turbolinks:load", function() {
  if (!$.fn.dataTable.isDataTable('#passengers')) { // more generic CSS selector
    var columns = [null, null, null, { "bSortable": false }, { "bSortable": false },  null,  { "bSortable": false }, { "bSortable": false }];
    if($(this).hasClass('.dispatch-table')
      columns.push({ "bSortable": false }); // some comment explaining why
    $('#passengers').DataTable({
      "aoColumns": columns,
      "aLengthMenu": [[25, 50, 75, 100, -1], [25, 50, 75, 100, "All"]],
      "iDisplayLength": 100
    });
  }
});
```

As far as I can tell, the only difference between the two is the extra element in `aoColumns` for the dispatch table. This is one way you could reflect that more concisely.

**Comment by dfaulken** [comment]
Probably just the JS file is fine, not the coffeescript file, correct?

**Comment by Chauce** [comment]
the coffeescript was to prevent the table from trying to reload on logout, iirc.

**Comment by dfaulken** [comment]
OK. Could you walk me through how it does that?

**Comment by dfaulken** [comment]
Duplicate code with #91?

**Comment by Chauce** [comment]
I derped - the coffeescript was the fix for dumplicate headers and footers appearing. It was a fix suggested off the Datatables plugin help forum. However ... I tried nuking the file and I didn't have the duplication error. So now I am not actually sure what it's doing.

**Comment by Chauce** [comment]
Closing this PR, because PR #97 will close both issues #58 and #96.

**Pull request closed without merging**